apps/src/templates/courseOverview/TeacherCourseOverview.tsx

```diff
@@ -116,7 +116,11 @@ const TeacherCourseOverview: React.FC = () => {
   const selectedSection = useAppSelector(selectedSectionSelector);
 
   React.useEffect(() => {
-    if (!selectedSection || !selectedSection?.courseVersionName) {
+    if (
+      !selectedSection ||
+      !selectedSection?.courseVersionName ||
+      parseInt(params.sectionId || '-1') !== selectedSection.id
```

Comment:
This is necessary because the selected section can change before the router changes the page. This guarantees that we won't unnecessarily navigate away or load until the selected section is correct.